frontend/src/pages/storageClasses/ResetCorruptConfigValueAlert.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { Button } from '@patternfly/react-core';
+import { SyncAltIcon } from '@patternfly/react-icons';
+
+import { StorageClassConfig, StorageClassKind } from '~/k8sTypes';
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+import { CorruptedMetadataAlert } from './CorruptedMetadataAlert';
+
+interface ResetCorruptConfigValueAlertProps {
+  storageClassName: string;
+  storageClassConfig: StorageClassConfig;
+  refresh: () => Promise<void | StorageClassKind[]>;
+}
+
+export const ResetCorruptConfigValueAlert: React.FC<ResetCorruptConfigValueAlertProps> = ({
+  storageClassName,
+  storageClassConfig,
+  refresh,
+}) => {
+  const [error, setError] = React.useState<string>();
+
+  const onClick = React.useCallback(async () => {
+    try {
+      await updateStorageClassConfig(storageClassName, storageClassConfig);
+      refresh();
+    } catch {
+      setError('Failed to refresh the field. Try again later.');
+    }
+  }, [storageClassName, storageClassConfig, refresh]);
+
+  return (
+    <CorruptedMetadataAlert
+      popoverText="Refresh the field to correct the corrupted metadata."
+      errorText={error}
+      action={
+        <Button
+          variant="plain"
+          aria-label="Corrupt metadata default reset button"
+          onClick={onClick}
+        >
+          <SyncAltIcon />
```

Comment:
this can be spammed, maybe add a isLoading to the disabled prop

Reply:
Good idea. Updated.